### PR TITLE
Ensure SNI flag set on multi-site hostname listener (Azure#426)

### PR DESCRIPTION
**Pull request by akshaysngupta**
This PR add support for `RequireServerNameIndication` field for V1 gateway.
This fields is needs to provide the correct certificate bound to the listener/hostname/port on V1.

### CODE REVIEW

File: pkg/appgw/public_and_private_ip_test.go [hunk 1]
```diff
@@ -333,208 +308,214 @@ var _ = Describe("Tests `appgw.ConfigBuilder`", func() {
...
-    --            }
-    --        ],
-    --        "urlPathMaps": null
-    --    },
-    --    "tags": {
-    --        "ingress-for-aks-cluster-id": "/subscriptions/subid/resourcegroups/aksresgp/providers/Microsoft.ContainerService/managedClusters/aksname",
-    --        "managed-by-k8s-ingress": "a/b/c"
-    --    }
-    --}`
+--    "properties": {
```
Comment (delqn) [inline pkg/appgw/public_and_private_ip_test.go]: The indentation changes make it really tough to see what changed in the output JSON.

Would you be able to either keep indentation as it was before this PR or first change the indentation in another PR and merge in here.  It'd be good to see what chunks of the App Gwy config actually changed.

File: pkg/appgw/configbuilder_test.go [hunk 2]
```diff
@@ -368,12 +362,17 @@ var _ = Describe("Tests `appgw.ConfigBuilder`", func() {
 --                        "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/backendHttpSettingsCollection/bp---namespace-----service-name---80-80---name--"
 --                    },
 --                    "httpListener": {
---                        "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/httpListeners/fl-foo.baz-80"
+--                        "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/httpListeners/fl-foo.baz-80"
```
Comment (delqn) [inline pkg/appgw/configbuilder_test.go]: A bunch of extra space at the end here

File: pkg/appgw/appgw_test.go [hunk 3]
```diff
@@ -283,7 +283,7 @@ var _ = Describe("Tests `appgw.ConfigBuilder`", func() {
 			Name: &listenerName,
 			ID:   to.StringPtr(appGwIdentifier.listenerID(listenerName)),
 			ApplicationGatewayHTTPListenerPropertiesFormat: &n.ApplicationGatewayHTTPListenerPropertiesFormat{
-				FrontendIPConfiguration: resourceRef("*"),
+				FrontendIPConfiguration: resourceRef("--front-end-ip-id-1--"),
```
Comment (delqn) [inline pkg/appgw/appgw_test.go]: Would you be so kind to use the `tests.PublicIPID` const?

File: pkg/appgw/appgw_test.go [hunk 4]
```diff
@@ -662,7 +647,7 @@ var _ = Describe("Tests `appgw.ConfigBuilder`", func() {
 					Name: &httpsListenerName,
 					ID:   to.StringPtr(appGwIdentifier.listenerID(httpsListenerName)),
 					ApplicationGatewayHTTPListenerPropertiesFormat: &n.ApplicationGatewayHTTPListenerPropertiesFormat{
-						FrontendIPConfiguration: resourceRef("*"),
+						FrontendIPConfiguration: resourceRef("--front-end-ip-id-1--"),
```
Comment (delqn) [inline pkg/appgw/appgw_test.go]: It'd be nice to reuse the `tests.PublicIPID` const.

**Comment by robertprather** [comment]
This is a highly desired change for us. Thanks team!

**Pull request merged**